File: 刷题笔记/2022/Online Judge/等差数列.py
def gcd(a, b):
    if b:
        return gcd(b, a % b)
    else:
        return a


def main():
    n = int(input())
    list = input()
    list = [int(i) for i in list.split()]

    list.sort()

    md = list[-1] - list[0]
    d = md
    for i in range(n - 1):
        d = gcd(d, list[i + 1] - list[i])

    if d == 0:
        print(n)
    else:
        print(md // d + 1)


if __name__ == '__main__':
    main()

File: 刷题笔记/2022/Online Judge/等差数列.cpp
// http://acm.hgnu.edu.cn/p/167
#include <iostream>
#include <algorithm>

using namespace std;

int gcd(int a, int b) {
    return b ? gcd(b, a % b) : a;
}

int main(){
	int n;
	cin >> n;
	int a[n];
	for(int i = 0; i < n; i++){
		cin >> a[i];
	}
	sort(a, a + n);
	int mina = a[1] - a[0];
	int maxa = (a[n - 1] - a[0]);
	for(int i = 0; i < n - 1; i++){
		int temp = a[i + 1] - a[i];
		mina = gcd(maxa, temp);
	}
	if(mina == 0) cout << n;
	else{
		int ans = maxa / mina + 1;
		cout << ans;
	}
	return 0;
}
